### DISCUSSION TITLE
AppCheck

### DISCUSSION THREAD

**Discussion by botanarede** [enesien/venefish#6] [Ideas]
I needed to use this app check provider to be able to use an application that I already had running with app check:

**firebase-app-check-provider.tsx**

> 'use client'
> 
> import { AppCheckProvider, useFirebaseApp } from 'reactfire'
> import { initializeAppCheck, ReCaptchaV3Provider } from 'firebase/app-check'
> // import { isBrowser } from 'src/lib/generic/isBrowser'
> 
> // const FirebaseAppCheckProvider: React.FCC = ({ children }) => {
> const FirebaseAppCheckProvider = ({ children }: any) => {
>   const pageConfiguration = {
>     emulator: process.env.NEXT_PUBLIC_EMULATOR === 'true',
>     appCheckDebugToken: process.env.NEXT_PUBLIC_APP_DEV === 'true',
>   }
> 
>   const siteKey = process.env.NEXT_PUBLIC_FIREBASE_APP_CHECK
> 
>   const app = useFirebaseApp()
> 
>   if (!siteKey || !isBrowser() || pageConfiguration.emulator) {
>     return <>{children}</>
>   }
> 
>   // if (pageConfiguration.appCheckDebugToken) {
>   // attachAppCheckDebugToken()
>   // }
> 
>   const provider = new ReCaptchaV3Provider(siteKey)
> 
>   const sdk = initializeAppCheck(app, {
>     provider,
>     isTokenAutoRefreshEnabled: true,
>   })
> 
>   return <AppCheckProvider sdk={sdk}>{children}</AppCheckProvider>
> }
> 
> export default FirebaseAppCheckProvider
> 
> // function attachAppCheckDebugToken() {
> //   const token = process.env.NEXT_PUBLIC_FIREBASE_APPCHECK_DEBUG_TOKEN
> 
> //   Object.assign(window, {
> //     FIREBASE_APPCHECK_DEBUG_TOKEN: token,
> //   })
> // }
> 
> function isBrowser() {
>   return typeof window !== 'undefined'
> }

and wrapped content

**layout.tsx**

> \<MyFirebaseProvider\>
>     \<FirebaseAppCheckProvider\>
>         {children}
>         \<Toaster /\>
>     \</FirebaseAppCheckProvider\>
> \</MyFirebaseProvider\>

has someone a better idea?

**Comment by ncast-dev** [comment]
## DIGI-453 — Test E2E Phase 0 pipeline smoke ✅

**Story:** Test E2E Phase 0 — 1 story bidon traverse tout le nouveau pipeline

**Résultat:** 1 test GREEN valide les 8 fixes Phase 0 (DIGI-441..448) :
- Tous les modules Phase 0 importables (SpecWriter, git_verification, senior_engineer, code_reviewer)
- Pipeline complet simulé : Senior crée PR → Reviewer APPROVE → SHA vérifié sur main → story done
- LLM calls mockés (zéro API), git_verification avec vrai repo local

**Commit:** `aa4eadb` sur `feature/DIGI-449-branch-cleanup`

**Tests:** 1 passed (pre-existing failure test_DIGI-311 non liée à ce changement)
